src/macosx/osxgl.m

```diff
@@ -1644,15 +1631,14 @@ static void init_new_vsync(ALLEGRO_DISPLAY_OSX_WIN *dpy)
       }
       [fmt release];
       [view release];
-
+      if (display->flags & ALLEGRO_FULLSCREEN_WINDOW) {
+           // TODO fix main thread
```

Comment:
Could you clarify what needs to be done here? is this not running on the right thread?

Reply:
It was a note to myself. I fixed it (by moving that whole block up inside the `dispatch_sync`) but didn't delete the note. Sorry about that, I've pushed a new commit to get rid of it.